bug/#274 improve doc on job cancellation (#189)
Avaiga/taipy-doc: docs/manuals/core/entities/scheduling-and-job-execution.md

> -   `BLOCKED`: The job is blocked by inputs not ready.

Reviewer: "blocked by inputs not ready"
->
"blocked because inputs are not ready"

-   `PENDING`: The job is waiting for execution.
-   `RUNNING`: The job is being executed.

> -   `CANCELED`: The job was canceled by the user.

Reviewer: Only the user can cancel a Job?
Shouldn't this be "The job was canceled."?

Author: Yes only the user can cancel the job. The subsequent job of the canceled job that will not run is called ABANDONED

Jobs are created when a task is submitted.

> - You can cancel a Job with `taipy.cancel_job(job)^`. When canceling a Job, you will effectively set the `Status^` of subsequent jobs of the canceled Job to `ABANDONED`. However, a job cannot be canceled if it is already running. When canceling a running job, its subsequent jobs will be abandoned and the running job will continue running until completed.

Reviewer: Please decide whether you want to use 'job' or 'Job'.
This mix makes no sense.

Author: I decided to go with 'job' :D

> - You can cancel the Job with `taipy.cancel_job(job)^`. When cancelling a Job, you will effectively set the `Status^` of subsequent jobs of the cancelled Job to `ABANDONED`. However, if a job is already being run, it cannot be cancelled. When cancelling a running job, its subsequent jobs will be abandoned and the running job will continue running until completed.

Reviewer 2: "Cancelling" is British English while "Canceling" is more international English. We tend to use International English.

```suggestion
- You can cancel a Job with `taipy.cancel_job(job)^`. When canceling a Job, you will effectively set the `Status^` of subsequent jobs of the canceled Job to `ABANDONED`. However, a job cannot be canceled if it is already running. When canceling a running job, its subsequent jobs will be abandoned and the running job will continue running until completed.
```

> !!! example "Cancelling a job"

Reviewer 2: ```suggestion
!!! example "Canceling a job"
```

>     ```python linenums="1"

Reviewer: Please use the line numbering if and only if you reference a specific line or lines range.
This makes the reading better.

    import taipy as tp

    def double(nb):
        sleep(5)
        return nb * 2

    print(f'(1) Number of jobs: {len(tp.get_jobs())}.')

    # Create a scenario then submit it.
    input_data_node_config = tp.configure_data_node("input", default_value=21)
    output_data_node_config = tp.configure_data_node("output")
    double_task_config = tp.configure_task("double_task", double, input_data_node_config, output_data_node_config)
    print_task_config = tp.configure_task("print_task", print, output_data_node_config)
    scenario_config = tp.configure_scenario_from_tasks("my_scenario", [double_task_config, print_task_config])
    scenario = tp.create_scenario(scenario_config)
    tp.submit(scenario)

>     # Retrieve all jobs.

Reviewer: This comment should be something like:
    Count the Jobs.
    
 (as you have used capitals in the whole text)

Author: I'll go for "Count the jobs" :D

    print(f'(2) Number of jobs: {len(tp.get_jobs())}.')

>     # Get the latest created job of a Task.

Reviewer: Please drop this useless comment

    jobs = tp.get_latest_job(scenario.double_task)

>     # Get status of the Job

Reviewer: ... of the Jobs.

    print(f'(3) Status of job double_task: {job[0].status}')
    print(f'(4) Status of job print_task: {jobs[1].status}')

>     # Then cancel it.

Reviewer: "Cancel the second Job."

    tp.cancel_job(job[1])

    sleep(10)

    print(f'(5) Status of job double_task: {job[0].status}')
    print(f'(6) Status of job print_task: {jobs[1].status}')
    ```

This example will produce the following output:

```
(1) Number of jobs: 0.
(2) Number of jobs: 2.
(3) Status of job double_task: Status.RUNNING
(4) Status of job print_task: Status.BLOCKED
(5) Status of job double_task: Status.COMPLETED

> (6) Status of job print_task: Status.CANCELLED

Reviewer: So I guess this 'CANCELLED' symbol should be... 'CANCELED' right?

Author: seems so to me, I'll make another PR to fix that in taipy-core :D

```

> !!! example "Cancelling a running job"

Reviewer 2: ```suggestion
!!! example "Canceling a running job"
```

    ```python
    import taipy as tp

    def double(nb):
        sleep(5)
        return nb * 2

    print(f'(1) Number of jobs: {len(tp.get_jobs())}.')

    # Create a scenario then submit it.
    input_data_node_config = tp.configure_data_node("input", default_value=21)
    output_data_node_config = tp.configure_data_node("output")
    double_task_config = tp.configure_task("double_task", double, input_data_node_config, output_data_node_config)
    print_task_config = tp.configure_task("print_task", print, output_data_node_config)
    scenario_config = tp.configure_scenario_from_tasks("my_scenario", [double_task_config, print_task_config])
    scenario = tp.create_scenario(scenario_config)
    tp.submit(scenario)

>     # Retrieve all jobs.

Reviewer: Same comments as on the (looking identical) code before...

    print(f'(2) Number of jobs: {len(tp.get_jobs())}.')

    # Get the latest created job of a Task.
    jobs = tp.get_latest_job(scenario.double_task)

    # Get status of the Job
    print(f'(3) Status of job double_task: {job[0].status}')
    print(f'(4) Status of job print_task: {jobs[1].status}')

    # Then cancel it.
    tp.cancel_job(job[0])

    sleep(10)

    print(f'(5) Status of job double_task: {job[0].status}')
    print(f'(6) Status of job print_task: {jobs[1].status}')
    ```

> This example will produce the following output:

Reviewer: Use present if that's what happens:

"This example produces the following..."
